app/controllers/documents_controller.rb

```diff
@@ -125,4 +134,37 @@ def prepare_for_mobile
     session[:mobile_param] = params[:mobile] if params[:mobile]
    # request.format = :mobile if mobile_device?
   end
+
+  private
+
+  def catalog_texts
+
+    if catalogue_enabled?
+       status, results = Melcatalog.texts
+       return results[:text] unless results[:text].nil?
+    end
+    return []
+  end
+
+  def catalog_content( doc )
+
+    if catalogue_enabled?
+      # we put placeholder content in earlier and replace with the real thing now
+      if doc.text.start_with?( "EID:" )
+         eid = doc.text.split( ":",2 )[ 1 ]
+         status, entry = Melcatalog.get( eid, 'stripxml' )
+         if status == 200 && entry && entry[:text] && entry[:text][ 0 ] && entry[:text][ 0 ]['content']
+           doc.text = entry[:text][ 0 ]['content']
+         else
+           doc.text = "Error getting document content from the catalog; status = #{status}, eid = #{eid}"
+         end
+      end
+    end
+  end
+
+  # helper to determine if we should support content from the MEL catalog
+  def catalogue_enabled?
```

Comment:
On second thought, there is a reasonably easy, if rather kludgy, solution, by adding the following to the annotator setup:

```Javascript

  var optionsRichText = {
    editor_enabled: '<%= ENV["ANNOTATOR_RICHTEXT"] %>',
    tinymce: {
      <% if ENV["CATALOG_ENABLED"] == 'true' %>
      'toolbar': '<%= ENV["ANNOTATOR_RICHTEXT_WITH_CATALOG"] %>',
      <% end %>
      'toolbar': '<%= ENV["ANNOTATOR_RICHTEXT_CONFIG"] %>',
      'image_dimensions': false
    }
  };

```

And creating alternate config vars for tinymce.